File: app/src/main/java/com/akameko/testforlifehackstudio/ui/main/DetailFragment.java
package com.akameko.testforlifehackstudio.ui.main;

import android.content.res.AssetManager;
import android.graphics.drawable.Drawable;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.akameko.testforlifehackstudio.R;
import com.akameko.testforlifehackstudio.repository.CompanyDetailItem;
import com.akameko.testforlifehackstudio.repository.CompanyListItem;

import java.io.IOException;
import java.io.InputStream;


public class DetailFragment extends Fragment {
    private SharedViewModel sharedViewModel;
    private CompanyDetailItem companyDetailItem;

    TextView companyName;
    TextView companyPhone;
    TextView companyWww ;
    TextView companyMap ;
    TextView companyDescription ;
    ImageView companyPhoto;

    public DetailFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View root = inflater.inflate(R.layout.fragment_detail, container, false);
        companyName = root.findViewById(R.id.text_view_details_name);
        companyPhone = root.findViewById(R.id.text_view_details_phone);
        companyWww = root.findViewById(R.id.text_view_details_www);
        companyMap = root.findViewById(R.id.text_view_details_map);
        companyDescription = root.findViewById(R.id.text_view_details_description);
        companyPhoto = root.findViewById(R.id.image_view_details);
       return root;
    }

    @Override
    public void onStart() {
        super.onStart();

        sharedViewModel = new ViewModelProvider(requireActivity()).get(SharedViewModel.class);
        sharedViewModel.getDetails().observe(getViewLifecycleOwner(), details -> {

            for (CompanyDetailItem c : details) {
                Log.d("", c.toString());
            }
            companyDetailItem = details.get(0);
            initViews();
        });

    }

    private void initViews(){
        companyName.setText(companyDetailItem.getName());
        if (!companyDetailItem.getPhone().equals("")){
            companyPhone.setText(companyDetailItem.getPhone());
            companyPhone.setVisibility(View.VISIBLE);
        }
        if (!companyDetailItem.getWww().equals("")){
            companyWww.setText(companyDetailItem.getWww());
            companyWww.setVisibility(View.VISIBLE);
        }
        if (!companyDetailItem.getLat().equals("0") && !companyDetailItem.getLon().equals("0")){
            companyMap.setText(companyDetailItem.getLat() + ", "+ companyDetailItem.getLon());
            companyMap.setVisibility(View.VISIBLE);
        }
        if (!companyDetailItem.getDescription().equals("")){
            companyDescription.setText(companyDetailItem.getDescription());
            companyDescription.setVisibility(View.VISIBLE);
        }






        AssetManager am = getResources().getAssets();
        try {
            InputStream is = am.open(companyDetailItem.getImg());
            companyPhoto.setImageDrawable( Drawable.createFromStream(is, String.valueOf(0)));
        } catch (IOException e) {
            e.printStackTrace();
        }
    }


}

File: app/src/main/java/com/akameko/testforlifehackstudio/ui/main/DetailFragment.kt
package com.akameko.testforlifehackstudio.ui.main

import android.graphics.drawable.Drawable
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.akameko.testforlifehackstudio.R
import com.akameko.testforlifehackstudio.repository.CompanyDetailItem
import java.io.IOException

class DetailFragment : Fragment() {
    private var sharedViewModel: SharedViewModel? = null
    private var companyDetailItem: CompanyDetailItem? = null
    var companyName: TextView? = null
    var companyPhone: TextView? = null
    var companyWww: TextView? = null
    var companyMap: TextView? = null
    var companyDescription: TextView? = null
    var companyPhoto: ImageView? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? { // Inflate the layout for this fragment
        val root = inflater.inflate(R.layout.fragment_detail, container, false)
        companyName = root.findViewById(R.id.text_view_details_name)
        companyPhone = root.findViewById(R.id.text_view_details_phone)
        companyWww = root.findViewById(R.id.text_view_details_www)
        companyMap = root.findViewById(R.id.text_view_details_map)
        companyDescription = root.findViewById(R.id.text_view_details_description)
        companyPhoto = root.findViewById(R.id.image_view_details)
        return root
    }

    override fun onStart() {
        super.onStart()
        sharedViewModel = ViewModelProvider(requireActivity()).get(SharedViewModel::class.java)
        sharedViewModel!!.details.observe(viewLifecycleOwner, Observer { details: List<CompanyDetailItem> ->
            for (c in details) {
                Log.d("", c.toString())
            }
            companyDetailItem = details[0]
            initViews()
        })
    }

    private fun initViews() {
        companyName!!.text = companyDetailItem!!.name
        if (companyDetailItem!!.phone != "") {
            companyPhone!!.text = companyDetailItem!!.phone
            companyPhone!!.visibility = View.VISIBLE
        }
        if (companyDetailItem!!.www != "") {
            companyWww!!.text = companyDetailItem!!.www
            companyWww!!.visibility = View.VISIBLE
        }
        if (companyDetailItem!!.lat != "0" && companyDetailItem!!.lon != "0") {
            companyMap!!.text = companyDetailItem!!.lat + ", " + companyDetailItem!!.lon
            companyMap!!.visibility = View.VISIBLE
        }
        if (companyDetailItem!!.description != "") {
            companyDescription!!.text = companyDetailItem!!.description
            companyDescription!!.visibility = View.VISIBLE
        }
        val am = resources.assets
        try {
            val `is` = am.open(companyDetailItem!!.img)
            companyPhoto!!.setImageDrawable(Drawable.createFromStream(`is`, 0.toString()))
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}